Source/Core/DiscIO/FileSystemGCWii.cpp

```diff
@@ -270,6 +271,19 @@ void CFileSystemGCWii::InitFileSystem()
 	if (!Root.IsDirectory())
 		return;
 
+	static const u32 ARBITRARY_FILE_SYSTEM_SIZE_LIMIT = 2 * 1024 * 1024;
+	if (Root.m_FileSize > ARBITRARY_FILE_SYSTEM_SIZE_LIMIT)
+	{
+		// Without this check, Dolphin may crash by trying to allocate too much memory
+		// when loading the filesystems of certain corrupt disc images. 12 bytes
+		// (the size of a file entry) times 2 * 1024 * 1024 is equal to the size of
+		// main GC/Wii RAM (24 MiB), and no file system should use anywhere near
```

Comment:
I don't want to move the whole comment to `ARBITRARY_FILE_SYSTEM_SIZE_LIMIT` because it's mainly about the check, and I don't want to only move part of it because it would be awkward to write and read this explanation in two separated parts.